CMake build directories break every update

### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
% brew config
HOMEBREW_VERSION: 3.2.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: ab2118a402a44451b1d7559167b6956dc47cc19a
Last commit: 22 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: bf1142ae8f34ecf5c9d372c13d0b9635ce35fc51
Core tap last commit: 18 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit skylake
Clang: 13.0.0 build 1300
Git: 2.30.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.0-x86_64
CLT: 13.0.0.0.1.1628499445
Xcode: 13.0 => /Applications/Xcode-beta.app/Contents/Developer

% brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /opt/devkitpro/tools/bin/pkg-config
  /opt/devkitpro/tools/bin/x86_64-apple-darwin15-pkg-config

Warning: You are using macOS 12.
We do not provide support for this pre-release version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
pre-release version.


Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```


### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Incrementally build a CMake project after having updated CMake from Homebrew.

### What happened (include all command output)?

```
nhp@narshe NanoboyAdvance % cmake --build build
gmake: /usr/local/Cellar/cmake/3.21.1/bin/cmake: No such file or directory
gmake: *** [Makefile:243: cmake_check_build_system] Error 127
```

### What did you expect to happen?

It picks up the new version of CMake automatically/does not rely on the specific versioned CMake directory from /usr/local/Cellar.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
1. `brew install cmake`
2. Clone a project that uses CMake, and create a build directory like `cmake -B build`
3. `brew upgrade` when there's a new version of CMake, or install another version some other way
4. CMake-generated Makefiles now can't find CMake
```

--- a/Formula/c/cmake.rb
+++ b/Formula/c/cmake.rb
@@ -31,6 +31,11 @@ class Cmake < Formula
     depends_on "openssl@3"
   end
 
+  # Prevent the formula from breaking on version/revision bumps.
+  # Check if possible to remove in 3.32.0
+  # https://gitlab.kitware.com/cmake/cmake/-/merge_requests/9978
+  patch :DATA
+
   # The completions were removed because of problems with system bash
 
   # The `with-qt` GUI option was removed due to circular dependencies if
@@ -78,3 +83,26 @@ class Cmake < Formula
     refute_path_exists man
   end
 end
+
+__END__
+diff --git a/Source/cmSystemTools.cxx b/Source/cmSystemTools.cxx
+index 5ad0439c..161257cf 100644
+--- a/Source/cmSystemTools.cxx
++++ b/Source/cmSystemTools.cxx
+@@ -2551,7 +2551,7 @@ void cmSystemTools::FindCMakeResources(const char* argv0)
+     _NSGetExecutablePath(exe_path, &exe_path_size);
+   }
+   exe_dir =
+-    cmSystemTools::GetFilenamePath(cmSystemTools::GetRealPath(exe_path));
++    cmSystemTools::GetFilenamePath(exe_path);
+   if (exe_path != exe_path_local) {
+     free(exe_path);
+   }
+@@ -2572,7 +2572,6 @@ void cmSystemTools::FindCMakeResources(const char* argv0)
+   std::string exe;
+   if (cmSystemTools::FindProgramPath(argv0, exe, errorMsg)) {
+     // remove symlinks
+-    exe = cmSystemTools::GetRealPath(exe);
+     exe_dir = cmSystemTools::GetFilenamePath(exe);
+   } else {
+     // ???

--- a/Formula/c/cmake.rb
+++ b/Formula/c/cmake.rb
@@ -17,12 +17,13 @@ class Cmake < Formula
   end
 
   bottle do
-    sha256 cellar: :any_skip_relocation, arm64_sequoia: "8657731ef6748f043699a3f95ea6eccc8d27180e088871c3eec29d47273f1ba9"
-    sha256 cellar: :any_skip_relocation, arm64_sonoma:  "07e0e20eae8e728544f4063aea9d959083285b25a951a5e3ff090818a1f98fe5"
-    sha256 cellar: :any_skip_relocation, arm64_ventura: "d1077d6ba9f88e9c8aaaa73de9c7ea186e9095fe34946fccde333d639438403c"
-    sha256 cellar: :any_skip_relocation, sonoma:        "3a0ea704d3af00a79fa96bffa6bb3f9d4794399537fc8d08db8c76fd1871c86f"
-    sha256 cellar: :any_skip_relocation, ventura:       "524d914a598a9c3a82711ca2954a9576fd1074b6c7030c6d6bbd796c763e917e"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:  "f7a6df026fffd0d69dc53899fee231b969f8e8872063b76773d0ec975a24f09e"
+    rebuild 1
+    sha256 cellar: :any_skip_relocation, arm64_sequoia: "c34fe6c7f83913b147faf3970db99213d1197f64a2bb411b45b4020e80e004dc"
+    sha256 cellar: :any_skip_relocation, arm64_sonoma:  "d10dcc360253726cc3872666f2ea7bc830bfb4dc25d73c79e28b4458973e301f"
+    sha256 cellar: :any_skip_relocation, arm64_ventura: "d1500ca2cb91a2b56708c8b0ec6825edde8cc1afbdeaede2f7fca18c2ca42db3"
+    sha256 cellar: :any_skip_relocation, sonoma:        "6dc996c171867e69a7daa690e2101f983a56a9bc010a699775606d6c9a731ebb"
+    sha256 cellar: :any_skip_relocation, ventura:       "51c2b535f266689edfcc5bb4e595c09544ea4221deaeed559c63a26f73d1df26"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:  "556d7c78c3c297098e13889ecaa603bd786bfdc43fc748ec900b02454360219c"
   end
 
   uses_from_macos "ncurses"

